libpod/container.go

```diff
@@ -588,3 +589,30 @@ func (c *Container) Export(path string) error {
 func (c *Container) Commit() (*storage.Image, error) {
 	return nil, ErrNotImplemented
 }
+
+// Wait blocks on a container to exit and returns its exit code
+func (c *Container) Wait() (int32, error) {
+
+	err := wait.PollImmediateInfinite(1,
+		func() (bool, error) {
+			if !c.isStopped() {
+				return false, nil
+			} else { // nolint
+				return true, nil // nolint
+			} // nolint
+
+		},
+	)
+
+	if err != nil {
+		return 0, err
+	}
+	exitCode := c.state.ExitCode
+	return exitCode, nil
+}
+
+func (c *Container) isStopped() bool {
+	_ = c.syncContainer()
```

Comment:
Maybe pass the error through? The func() in PollImmediateInfinite can return an error and I'm pretty sure this would block forever if someone else deleted the container while we were waiting.

Reply:
Also, add c.lock.Lock() / c.lock.Unlock() here, need it to make sure we don't do unfortunate things to the database.

Reply:
fixed

Reply:
Sorry, should have been more clear - lock needs to be above syncContainer(). syncContainer can potentially do a database write if the container exited, which needs to be locked.

Reply:
fixed